dashboard/app/models/pd/workshop_survey_foorm_submission.rb

```diff
@@ -0,0 +1,93 @@
+# == Schema Information
+#
+# Table name: pd_workshop_survey_submissions
+#
+#  id                  :integer          not null, primary key
+#  foorm_submission_id :integer          not null
+#  user_id             :integer          not null
+#  pd_session_id       :integer
+#  pd_workshop_id      :integer          not null
+#  day                 :integer
+#  created_at          :datetime         not null
+#  updated_at          :datetime         not null
+#
+# Indexes
+#
+#  index_pd_workshop_survey_submissions_on_foorm_submission_id  (foorm_submission_id) UNIQUE
+#  index_pd_workshop_survey_submissions_on_pd_session_id        (pd_session_id)
+#  index_pd_workshop_survey_submissions_on_pd_workshop_id       (pd_workshop_id)
+#  index_pd_workshop_survey_submissions_on_user_id              (user_id)
+#
+
+class Pd::WorkshopSurveyFoormSubmission < ApplicationRecord
+  include Pd::WorkshopSurveyConstants
+
+  VALID_DAYS = {
+    LOCAL_CATEGORY => (0..5).to_a.freeze,
+    ACADEMIC_YEAR_1_CATEGORY => [1].freeze,
+    ACADEMIC_YEAR_2_CATEGORY => [1].freeze,
+    ACADEMIC_YEAR_3_CATEGORY => [1].freeze,
+    ACADEMIC_YEAR_4_CATEGORY => [1].freeze,
+    ACADEMIC_YEAR_1_2_CATEGORY => [1, 2].freeze,
+    ACADEMIC_YEAR_3_4_CATEGORY => [1, 2].freeze,
+    VIRTUAL_1_CATEGORY => [1].freeze,
+    VIRTUAL_2_CATEGORY => [1].freeze,
+    VIRTUAL_3_CATEGORY => [1].freeze,
+    VIRTUAL_4_CATEGORY => [1].freeze,
+    VIRTUAL_5_CATEGORY => [1].freeze,
+    VIRTUAL_6_CATEGORY => [1].freeze,
+    VIRTUAL_7_CATEGORY => [1].freeze,
+    VIRTUAL_8_CATEGORY => [1].freeze,
+    CSF_CATEGORY => CSF_SURVEY_INDEXES.values.freeze
+  }
+
+  belongs_to :foorm_submission, class_name: 'Foorm::Submission'
+  belongs_to :user
+  belongs_to :pd_session, class_name: 'Pd::Session'
+  belongs_to :pd_workshop, class_name: 'Pd::Workshop'
```

Comment:
Do we actually want WorkshopSurveyFoormSubmission to have a reference to Workshop, or should we do a delegation or through-association using Session? (Some discussion of options [here](https://stackoverflow.com/questions/4021322/belongs-to-through-associations))

Reply:
We do because session is optional--ex the pre-survey doesn't have a session associated with it

Reply:
:bulb: Ah, I understand now.  Thanks!